tests/trainer/test_ddp.py

```diff
@@ -138,7 +138,8 @@ def test_ddp(device: DeviceHparams, world_size: int, composer_trainer_hparams: T
     We assert that each of these tensors are different to ensure that 1) random seeding works properly,
     and 2) each ddp process is indeed getting different data.
     """
-    del world_size  # unused. Set via env variables
+    if world_size == 1 and any([e not in os.environ for e in ['NODE_RANK', 'LOCAL_RANK', 'LOCAL_WORLD_SIZE']]):
```

Comment:
The DDP tests should work even when DDP isn't configured, since they should default to just using a world_size = 1, node_rank = 0, local_rank = 0, local_world_Size = 1. So this test shouldn't be skipped; there's actually a bug somewhere.

Reply:
I think I found the issue -- we initialize `dist` even when the world_size is 1 in our trainer.

Since we don't support using DDP without using our `composer` launcher, perhaps we shouldn't be testing that combination anyway?

Reply:
Yup, that is one of the issues, and there are a few other issues that we have to address. This PR runs the tests without DDP, so it should fail until we fully iron out the non-ddp path. https://github.com/mosaicml/composer/pull/389
That PR should address any DDP issues in this PR.

Reply:
removed the skip